lib/convection/control/cloud.rb

```diff
@@ -114,6 +114,18 @@ def diff(to_stack, options = {}, &block)
           return
         end
 
+        errors = []
+        stacks.each_value do |stack|
+          if stack.error?
+            errors << stack.errors.collect { |x| x.exception.message }
+          end
+        end
+        errors = errors.uniq.flatten!
+        unless errors.empty?
+          block.call(Model::Event.new(:error, "Error(s) during stack diff #{errors.join(', ')}", :error), errors) if block
+          return
+        end
```

Comment:
Yes we should add that at some point. I see two types of errors here, one during stack initialization, one during the diff/converge operation. For diff we are covered if there is a error during stack init, for converges i think the opposite. I am focusing on fixing the specific issue in diffs at the minute. We will open issues, track the appropriate changes we should make for converges however they will be prioritized after the diff work is complete.